test/e2e/encryption_at_rest_test.go

```diff
@@ -0,0 +1,190 @@
+				},
+			},
+			[]cloudaccess.Role{
+				{
+					Name: utils.RandomName(awsRoleNameBase),
+					AccessRole: v1.CloudProviderAccessRole{
+						ProviderName: "AWS",
+					},
+				},
+			},
+		),
+	)
+})
+
+func encryptionAtRestFlow(userData *model.TestDataProvider, encAtRest v1.EncryptionAtRest, roles []cloudaccess.Role) {
+	By("Deploy Project with requested configuration", func() {
+		actions.PrepareUsersConfigurations(userData)
+		deploy.NamespacedOperator(userData)
+		actions.DeployProjectAndWait(userData, "1")
+	})
+
+	By("Add cloud access role (AWS only)", func() {
+		if len(roles) == 0 {
+			return
+		}
+
+		err := cloudaccess.CreateRoles(roles)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		for _, role := range roles {
+			userData.Resources.Project.WithCloudAccessRole(role.AccessRole)
+		}
+
+		actions.PrepareUsersConfigurations(userData)
+		actions.DeployProject(userData)
+
+		Eventually(func() bool {
+			return EnsureAllRolesCreated(*userData, len(roles))
+		}).WithTimeout(5*time.Minute).WithPolling(20*time.Second).Should(BeTrue(), "Cloud access roles are not created")
+
+		project, err := kube.GetProjectResource(userData)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		err = cloudaccess.AddAtlasStatementToRole(roles, project.Status.CloudProviderAccessRoles)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		Eventually(kube.GetProjectCloudAccessRolesStatus(userData), "2m", "20s").Should(Equal("True"), "Cloud Access Roles status should be True")
+	})
+
+	By("Create KMS", func() {
+		project, err := kube.GetProjectResource(userData)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		Expect(len(project.Status.CloudProviderAccessRoles)).NotTo(Equal(0))
+		aRole := project.Status.CloudProviderAccessRoles[0]
+
+		fillKMSforAWS(&encAtRest, aRole.AtlasAWSAccountArn, aRole.IamAssumedRoleArn)
+		fillVaultforAzure(&encAtRest)
+		fillKMSforGCP(&encAtRest)
+
+		userData.Resources.Project.WithEncryptionAtRest(encAtRest)
+		actions.PrepareUsersConfigurations(userData)
+		actions.DeployProject(userData)
+	})
+
+	By("Check Encryption at Rest status", func() {
+		_, err := kube.GetProjectResource(userData)
+		Expect(err).ShouldNot(HaveOccurred())
+	})
+}
+
+func fillKMSforAWS(encAtRest *v1.EncryptionAtRest, atlasAccountArn, assumedRoleArn string) {
+	if (encAtRest.AwsKms == v1.AwsKms{}) {
+		return
+	}
+
+	Expect(encAtRest.AwsKms.Region).NotTo(Equal(""))
+	awsAction := cloud.NewAwsAction()
+	CustomerMasterKeyID, err := awsAction.CreateKMS(config.AWSRegionUS, atlasAccountArn, assumedRoleArn)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(CustomerMasterKeyID).NotTo(Equal(""))
+
+	encAtRest.AwsKms.CustomerMasterKeyID = CustomerMasterKeyID
+}
+
+func fillVaultforAzure(encAtRest *v1.EncryptionAtRest) {
+	if (encAtRest.AzureKeyVault == v1.AzureKeyVault{}) {
+		return
+	}
+
+	// todo: fill in
+}
+
+func fillKMSforGCP(encAtRest *v1.EncryptionAtRest) {
+	if (encAtRest.GoogleCloudKms == v1.GoogleCloudKms{}) {
+		return
+	}
+
+	// todo: fill in
```

Comment:
Does it need to be completed, or can it be removed?

Reply:
Yeah, this is a placeholder for future Azure tests. Can stay as is for now, I think 🤔